nix/overlays/libfabric/0001-cuda-dlopen-correctly.patch

```diff
@@ -0,0 +1,82 @@
+From 1c37266a3654e8ea5468ae1b5bef567c5c7753af Mon Sep 17 00:00:00 2001
```

Comment:
This should not be in our repo.  We need to disagree and commit with the Subspace team on this patch.

Reply:
https://github.com/ofiwg/libfabric/pull/10365 has been open for some time.

Reply:
Doesn't change my opinion here.

Reply:
Removed the patch (now fetched from the PR itself).

Reply:
No, that's not what I meant.  What I meant was that I have strong objections to automating building a development environment that includes a patch that our customers don't have and that the Libfabric community doesn't want.  Whether the patch is in our code or pulled from a PR upstream doesn't change the fact that we're not developing against upstream, and so we're doing it wrong.

Reply:
It doesn't work without this patch.

```
libfabric:29950:1734038763::core:core:cuda_hmem_dl_init():492<warn> Failed to dlopen libcuda.so
libfabric:29950:1734038763::core:core:ofi_hmem_init():662<warn> Failed to initialize hmem iface FI_HMEM_CUDA: No data available
libfabric:27266:1734038763::core:core:cuda_hmem_dl_init():492<warn> Failed to dlopen libcuda.so
libfabric:27266:1734038763::core:core:ofi_hmem_init():662<warn> Failed to initialize hmem iface FI_HMEM_CUDA: No data available
libfabric:27265:1734038763::core:core:cuda_hmem_dl_init():492<warn> Failed to dlopen libcuda.so
libfabric:27265:1734038763::core:core:ofi_hmem_init():662<warn> Failed to initialize hmem iface FI_HMEM_CUDA: No data available
libfabric:30091:1734038763::core:core:cuda_hmem_dl_init():492<warn> Failed to dlopen libcuda.so
libfabric:30091:1734038763::core:core:ofi_hmem_init():662<warn> Failed to initialize hmem iface FI_HMEM_CUDA: No data available
libfabric:30091:1734038763::core:mr:ofi_monitors_init():222<info> Initializing memory monitor cuda
libfabric:30091:1734038763::core:mr:ofi_monitors_init():222<info> Initializing memory monitor cuda_ipc
libfabric:30091:1734038763::core:core:fi_param_get_():372<info> variable mr_cuda_cache_monitor_enabled=<not set>
libfabric:27261:1734038763::core:core:cuda_hmem_dl_init():492<warn> Failed to dlopen libcuda.so
libfabric:27261:1734038763::core:core:ofi_hmem_init():662<warn> Failed to initialize hmem iface FI_HMEM_CUDA: No data available
```

libfabric should open libcuda.so.1, this is how libcudart itself resolves libcuda.so:

```
strings /usr/local/cuda/lib64/libcudart.so.12 | grep cuda.so
libcuda.so.1
```

And required for libfabric to work under nix.

Reply:
The upstream PR merged today, so this is now just backporting that patch to 1.22amzn4.0 and I'm hoping that changes your stance somewhat.